native/src/seal/polyarray.cpp

```diff
@@ -52,6 +52,29 @@ namespace seal
         }
     }
 
+    PolynomialArray &PolynomialArray::operator=(const PolynomialArray &assign)
```

Comment:
operator= is *not* a copy constructor and it's strange to overload = without having one.

A copy constructor has the signature `MyThing(const MyThing& rhs)`. In this case, you'll do something like

```cpp
PolynomialArray::PolynomialArray(const PolynomialArray &rhs):
  poly_size(rhs.poly_size()),
  coeff_modulus_size(rhs.coeff_modulus_size()),
  coeff_modulus_(rhs.coeff_modulus_),
  poly_modulus_degree(poly_modulus_degree) {
  reserve(poly_size, poly_modulus_degree, coeff_modulus_);

  for (std::size_t i = 0; i < poly_size; i++) {
    const auto data_ptr = rhs.get_polynomial(i);
    insert_polynomial(i, data_ptr);
  }  
}
```

Reply:
Don't overload this operator, but instead just implement an actual copy constructor.

Reply:
IIRC, the default implementation of operator= is to call the copy or move constructor as appropriate.

I think the default move constructor (i.e. `MyObject(const MyObject &&rhs) const`, note the && Lvalue reference) does has the correct semantics, so no need to override (i.e. bitwise copy).

Reply:
I know it is not the copy constructor, I was just copying the other SEAL class implementations to match their style.

Reply:
For example, here is the ciphertext example:

```c++
        /**
        Creates a new ciphertext by copying a given one.

        @param[in] copy The ciphertext to copy from
        @param[in] pool The MemoryPoolHandle pointing to a valid memory pool
        @throws std::invalid_argument if pool is uninitialized
        */
        Ciphertext(const Ciphertext &copy, MemoryPoolHandle pool) : Ciphertext(std::move(pool))
        {
            *this = copy;
        }
```